wp-resources/plugins/sil-dictionary-webonary/webonary/Webonary_Parts_Of_Speech.php

```diff
@@ -0,0 +1,161 @@
+	 * @param array $selected_values
+	 */
+	public function __construct(string $current_lang_code, ?array $parts, array $selected_values)
+	{
+		$this->current_lang_code = $current_lang_code;
+		$this->selected_values = $selected_values;
+
+		if (is_null($parts)) {
+			$this->parts = $this->GetPartsOfSpeechFromWordpress();
+		} else {
+			$this->parts = $parts;
+
+			// pre-sort the list
+			usort($this->parts, function ($a, $b) {
+				return strcasecmp($a->name, $b->name);
+			});
+		}
+	}
+
+	public function GetDropdown(): string
+	{
+		$counter = 1;
+		$all_parts = __('All Parts of Speech', 'sil_dictionary');
+		$template = '<div class="pos-entry"><input type="checkbox" name="tax" id="pos-%1$s" class="form-control mr-2 pos-check" value="%2$s" %3$s>&nbsp;<label for="pos-%1$s">%4$s</label></div>';
+		$options = [];
+
+		$value = '';
+		$checked = (empty($this->selected_values) || in_array($value, $this->selected_values)) ? 'checked="checked"' : '';
+		$options[] = sprintf($template, $counter, $value, $checked, $all_parts);
+
+		foreach ($this->parts as $part) {
+
+			if (is_null($part->abbreviation) || $part->lang != $this->current_lang_code)
+				continue;
+
+			$counter++;
+			$value = $part->abbreviation;
+			$checked = in_array($value, $this->selected_values) ? 'checked="checked"' : '';
+			$options[] = sprintf($template, $counter, $value, $checked, $part->name);
+		}
+
+		$option_str = implode(PHP_EOL, $options);
+
+		$button_text = __('Parts of Speech', 'sil_dictionary');
+
+		return <<<HTML
+<script type="text/javascript">
+	document.addEventListener('DOMContentLoaded', function() {
+        
+        // un-check options that are excluded by the selected option
+        jQuery('.pos-check').on('click', function() {
+            if (!this.checked)
+                return;
+
+            if (this.id === 'pos-1') {
+                // un-check everything else if 'All parts of speech' was selected
+                jQuery('.pos-check:checked').not(this).prop('checked', false);
+            }
+            else {
+                // un-check 'All parts of speech' if something else was selected
+                jQuery('#pos-1:checked').prop('checked', false);
+            }
+        });
+	});
+</script>
+<div class="pos-container">
+	<button type="button" class="btn btn-dropdown" onclick="jQuery('.pos-list').toggle();">$button_text<span>&ensp;&#x25BE;</span></button>
+	<div class="pos-list" style="display: none">
+	    $option_str
+    </div>
+</div>
+HTML;
+	}
+
+	/**
+	 * Returns the selected parts of speech as an array of strings.
+	 *
+	 * NOTE: An empty array is 'All parts of speech'
+	 *
+	 * @return array
+	 */
+	public static function GetPartsOfSpeechSelected(): array
+	{
+		if (!is_null(self::$parts_of_speech_selected))
+			return self::$parts_of_speech_selected;
+
+		// NOTE: There may be multiple 'tax' values in the query string,
+		//       but the $_GET[] array will only contain one.
+		$tax = array_values(
```

Comment:
Looks like `$_GET` will work if the `name` of the `input`s are `tax[]`: https://stackoverflow.com/a/1833398/2503659

Reply:
@johntseng true, but doing it that way would break other pages. The `tax` name is standard WordPress usage.